spring-test/src/main/java/org/springframework/test/web/servlet/ResultMatcher.java

```diff
@@ -70,4 +73,29 @@ static ResultMatcher matchAll(ResultMatcher... matchers) {
 		};
 	}
 
+	/**
+	 * Static method for matching with an array of result matchers whose assertion failures are caught and stored.
+	 * Only when all of them would be called a {@link AssertionError} be thrown containing the error messages of those
+	 * previously caught assertion failures.
+	 * @param matchers the matchers
+	 * @author Michał Rowicki
+	 * @since 5.2
+	 */
+	static ResultMatcher matchAllSoftly(ResultMatcher... matchers) {
+		return result -> {
+			List<String> failedMessages = new ArrayList<>();
+			for (ResultMatcher matcher : matchers) {
+				try {
+					matcher.match(result);
+				}
+				catch (AssertionError assertionException) {
+					failedMessages.add(assertionException.getMessage());
```

Comment:
to be consistent with WebTest version you could consider adding indices here https://github.com/spring-projects/spring-framework/pull/26969/files#diff-396f20e4c7198aab834e4d234e312f65e9fedf27d32582f336d1e8b0ca0a0328R521

Reply:
good idea, fixed